### PR TITLE
Add an rpm spec file

### PR DESCRIPTION
**Pull request by Gundersanne**
Includes a dockerfile and a `make container` target which installs the rpms in a f31 container.

### CODE REVIEW

File: distribution/osbuild-installer.socket [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[Unit]
+Description=OSBuild Composer RCM API socket
```
Comment (teg) [inline distribution/osbuild-installer.socket:2]: Copy-paste error ;)

File: osbuild-installer.spec [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+# Upstream license specification: Apache-2.0
+License:	ASL 2.0
+URL:		%{gourl}
+Source0:        %{gosource}
+
+BuildRequires:	%{?go_compiler:compiler(go-compiler)}%{!?go_compiler:golang}
+BuildRequires:	systemd
+
+Requires: systemd
+Provides: osbuild-installer
```
Comment (teg) [inline osbuild-installer.spec:24]: This is not needed. A package always provides itself.

File: Makefile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-PACKAGE_NAME = osbuild-installer-backend
```
Comment (teg) [inline Makefile]: I like this change! Would it makes sense to rename the whole project to `osbuild-installer` and drop the `-backend` suffix?

Comment (Gundersanne) [reply Makefile]: it does sound a bit silly, I copied it from the cloud rh projects but yea I'm in favour of dropping it :)

Comment (Gundersanne) [reply Makefile]: Hm I don't have the rights to change the name of this project. Could you rename it to drop the `-backend` ?

**Pull request merged**